mdsuite/time_series/base.py

```diff
@@ -0,0 +1,70 @@
+"""
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+https://www.eclipse.org/legal/epl-v20.html
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincware Project.
+
+Description: A base class for time series analysis
+"""
+from __future__ import annotations
+from mdsuite.database.simulation_database import Database
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from mdsuite import Experiment
+
+
+class TimeSeries:
+    def __init__(self, experiment: Experiment):
+        """
+
+        Parameters
+        ----------
+        experiment: Experiment
+            The parent experiment class to perform the time series operation on
+        """
+        self.experiment = experiment
+
+        self.loaded_property = None
+        self.fig_labels = {
+            "x": None,
+            "y": None
+        }
+
+        # Properties
+        self._database = None
+        self._data = None
+
+    def __call__(self):
+        self.plot()
+
+    @property
+    def database(self):
+        """Get the database"""
+        if self._database is None:
+            self._database = Database(name=self.experiment.database_path + r"\database.hdf5")
+        return self._database
+
+    @property
+    def data(self):
```

Comment:
I think a species argument would be better. Add species as a parameters and set all species as default.